Lib/test/test_unittest/testmock/testmock.py

```diff
@@ -1652,6 +1652,19 @@ def test_mock_unsafe(self):
         m.aseert_foo_call()
         m.assrt_foo_call()
 
+    # gh-100739
+    def test_mock_safe_with_spec(self):
+        class Foo(object):
+            def assert_bar(self):
+                pass
+
+            def assertSome(self):
+                pass
+
+        m = Mock(spec=Foo)
+        m.assert_bar()
+        m.assertSome()
```

Comment:
It wouldn't hurt here to also verify that child mocks are created here as normal (i.e. just verify the calls to them are recorded using `assert_called`?), rather than just that you can access the attribute without error.